Utils/azuretools-core/src/com/microsoft/azuretools/telemetrywrapper/EventUtil.java

```diff
@@ -28,29 +28,107 @@
         }
-        properties.put(CommonUtil.OPERATION_NAME, operName);
-        properties.put(CommonUtil.OPERATION_ID, UUID.randomUUID().toString());
-        sendTelemetry(eventType, eventName, mergeProperties(properties), metrics);
     }
 
-    public static void logError(String eventName, String operName, ErrorType errorType, String errMsg,
+    public static void logEvent(EventType eventType, String eventName, String operName,
+        Map<String, String> properties) {
+        logEvent(eventType, eventName, operName, properties, null);
+    }
+
+    public static void logError(String eventName, String operName, ErrorType errorType, Exception e,
+        Map<String, String> properties, Map<String, Double> metrics) {
+        try {
+            if (properties == null) {
+                properties = new HashMap<>();
+            }
+            properties.put(CommonUtil.OPERATION_NAME, operName);
+            properties.put(CommonUtil.OPERATION_ID, UUID.randomUUID().toString());
+            properties.put(CommonUtil.ERROR_CODE, "1");
+            properties.put(CommonUtil.ERROR_MSG, e != null ? e.getMessage() : "");
+            properties.put(CommonUtil.ERROR_CLASSNAME, e != null ? e.getClass().getName() : "");
+            properties.put(CommonUtil.ERROR_TYPE, errorType.name());
+            sendTelemetry(EventType.error, eventName, mergeProperties(properties), metrics);
+        } catch (Exception ignore) {
+        }
+    }
+
+    public static void logEvent(EventType eventType, Operation operation, Map<String, String> properties,
+        Map<String, Double> metrics) {
+        ((DefaultOperation) operation).logEvent(eventType, properties, metrics);
+    }
+
+    public static void logEvent(EventType eventType, Operation operation, Map<String, String> properties) {
+        logEvent(eventType, operation, properties, null);
+    }
+
+    public static void logError(Operation operation, ErrorType errorType, Exception e,
         Map<String, String> properties, Map<String, Double> metrics) {
-        if (properties == null) {
-            properties = new HashMap<>();
+        ((DefaultOperation) operation).logError(errorType, e, properties, metrics);
+    }
+
+    public static void logCommand(String eventName, String operName, Map<String, String> properties,
+        Map<String, Double> metrics, TelemetryConsumer<Operation> command, Consumer<Exception> errorHandle) {
+        Operation operation = TelemetryManager.createOperation(eventName, operName);
+        try {
+            operation.start();
+            command.accept(operation);
+        } catch (Exception e) {
+            logError(operation, ErrorType.userError, e, properties, metrics);
+            if (errorHandle != null) {
+                errorHandle.accept(e);
+            }
+        } finally {
+            operation.complete();
+        }
+    }
+
+    public static <R> R logCommand(String eventName, String operName, Map<String, String> properties,
```

Comment:
logCommand is not applicable in this case, since you have executed the method.

Reply:
Just rename to executeWithLog